test/Chainweb/Test/Mempool/InMem.hs

```diff
@@ -16,12 +18,19 @@ import Chainweb.Utils (Codec(..))
 tests :: TestTree
 tests = testGroup "Chainweb.Mempool.InMem"
             $ Chainweb.Test.Mempool.tests
-            $ MempoolWithFunc
-            $ InMem.withInMemoryMempool cfg
+            $ MempoolWithFunc wf
   where
+    wf f = do
+        mv <- newMVar (\v -> V.mapM (const $ return True) v)
+        let cfg = InMemConfig txcfg mockBlockGasLimit 2048 (checkMv mv)
+        InMem.withInMemoryMempool cfg $ f mv
+
+    checkMv mv xs = do
+        f <- readMVar mv
+        f xs
```

Comment:
`f <$> readMVar mv`

Reply:
Ehhhhh, check again :)

The closest pointfree I got was `readMVar mv >>= ($ xs)` but that's soup

Reply:
Oh my bad, you're right. Personally I'd drink that soup, but not all would.